server/src/termdb.gdc.js

```diff
@@ -936,12 +1006,45 @@ async function fetchIdsFromGdcApi(ds, size, from, aliquot_id) {
 					for (const aliquot of analyte.aliquots) {
 						const aliquot_id = aliquot.aliquot_id
 						if (!aliquot_id) throw 'aliquot.aliquot_id missing'
-						ds.aliquot2submitter.cache.set(aliquot_id, sample_submitter_id)
-						ds.map2caseid.cache.set(aliquot_id, case_id)
+						ds.__gdc.aliquot2submitter.cache.set(aliquot_id, sample_submitter_id)
+						ds.__gdc.map2caseid.cache.set(aliquot_id, case_id)
 					}
 				}
 			}
 		}
 	}
 	return re.data?.pagination?.total
 }
+
+async function checkExpressionAvailability(ds) {
+	// hardcodes this url since the api is only in uat now. replace with path.join() when it's in prod
+	const url = 'https://uat-portal.gdc.cancer.gov/auth/api/v0/gene_expression/availability'
+
+	{
+		/*
+		when the api is released in production, delete this check and move it into testGDCapi()
+		exp api is only available on uat now, thus it works if the pp server ip is whitelisted, otherwise won't work and will skip this function
+		*/
+		const c = await testRestApi(url)
+		if (c == 'ERR_NON_2XX_3XX_RESPONSE') {
```

Comment:
@siosonel i use error code with string value to detect if the api is accessible. as i saw that from a non-whitelisted IP it returns a different code. do you think it's okay?